---
layout: page
title: Monoall-ms
---

<link rel="stylesheet" href="{{ "/css/manuscript-summary.css" | prepend: site.baseurl }}">

<h2>Aging and genomic imprinting in the human brain</h2>

<section>
    <h3>Preliminaries</h3>
    <h4>Notation</h4>
    <p><strong>Color key</strong>
        <ul>
            <li> <dfn>normal color:</dfn> text related to main figure</li>
            <li> <dfn class=supplementary>first color:</dfn> ...supplementary figure</li>
            <li> <dfn class=missing>second color:</dfn> ...figure still missing</li>
        </ul>
    </p>
    <p> <strong>refname:</strong> this should be used as unique identifier for
    a figure until the final figure numbers may be established</p>
    <h4>Definitions</h4>
    <ul>
        <li>
            <dfn>monoallelic expression:</dfn> only one allele is expressed to a
            certain degree
        </li><li>
            <dfn>(genomic) imprinting:</dfn> a subtype of monoallelic expression
            where the same allele is expressed in some tissue across all
            individuals
        </li><li>
            <dfn>known imprinted</dfn> and <dfn>candidate</dfn> genes: with
            and without prior evidence for imprinting, respectively
        </li><li>
            <dfn>candidate, \lt 1 MB gene:</dfn> a candidate gene in the 1 MB
            neighborhood of some known imprinted gene
        </li><li>
            <dfn>imprinted gene cluster:</dfn> a sequence of neighboring genes
            that fall in either the "known" or the "candidate, \lt 1 MB gene"
            category
        </li><li>
            <dfn>biallelic expression:</dfn> both alleles are expressed to a
            certain degree
        </li><li>
            <dfn>unbiased, or balanced, expression:</dfn> the two alleles are
            expressed at equally high level
        </li><li>
            <dfn>parental expression bias</dfn> or simply <dfn>parental bias:</dfn> one allele is expressed at a
            higher level
        </li><li>
            <dfn>mean transcript ratio, \mu_{ig}</dfn> of a given individual i
            and gene g:
            <ol>
                <li>the mean ratio of the number of maternal transcripts in
                    all transcripts over a suitable period of time and set of
                    cells
                </li><li>hence a natural but not directly unobservable measure of expression bias</li>
            </ol>
        </li>
    </ul>
</section>

<section>
    <h3>Parental expression bias: variation across genes and individuals</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-07-19-genome-wide-S/complex-plot-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-07-19-genome-wide-S/complex-plot-1.png">
        </a>
        <figcaption>
        ranking-genes
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-07-19-genome-wide-S/complex-plot-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
        <strong>Motivation:</strong>
        Genome-wide and population-wide quantification parental bias
        <ol>
            <li>variability across genes and the fraction of monoallelically
                expressed genes</li>
            <li>variability across individuals</li>
        </ol>
        We define<dfn>read count ratio statistic, S_{ig}</dfn> of a given
        individual i and gene g as
        the greater of the observed read count of the two alleles at each
        heterozygous SNP summed over all such SNPs and normalized to the
        corresponding total read count.  We use S_{ig} to quantify parental
        bias.  This is justified by the reasonable assumptions that
        <ol>
            <li>the number of paternal and maternal transcripts are Poisson
                distributed with rate \lambda and \lambda \mu, respectively
            </li>
            <li>the observed paternal and maternal read count are both
                Poisson distributed with rate proportional to the number of
                paternal and maternal transcripts, respectively.
            </li>
        </ol>
        We also define the <dfn>gene g's score</dfn> as the fraction of
        individuals i for which S_{ig}\gt 0.9.  This is the same as 1 -
        ECDF(0.9), where ECDF is the empirical cumulative distribution
        function based on observations S_{1g},...,S_{Ig} on I individuals or,
        equivalently, the same as the corresponding survival function.
    </p>
    <p>
        <strong>Result:</strong>
        Three typical genes are shown in the <strong>upper 3 panels</strong>:
        PEG10 and ZNF331 are known imprinted genes, whereas AFAP1 is a
        candidate gene.  The each gene has a characteristic survival function
        1 - ECDF(s) based on which receive different scores <strong>green
            circles a, b, c</strong>.  The survival functions are color-coded
        for terse representation.  All ca. 5K genes that passed our quality
        filters are shown in this representation<strong>2 main lower
            panels</strong>; they are ranked according to their score, shown
        in the auxiliary <strong>right lower panels</strong>.
    </p>
    <p>
        <strong>Interpretation:</strong>
        Given some reasonable assumptions (see definition of read count ratio
        statistic S) only about ca. 1 % of the genes are expressed
        monoallelically
    </p>
    <p>
        <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-07-19-genome-wide-S %}">
            2016-07-19-genome-wide-S
        </a>
    </p>
</section>

<section>
    <h3>Parental bias and imprinted gene clusters</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/score-genomic-location-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/score-genomic-location-1.png">
        </a>
        <figcaption>
        clusters
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/score-genomic-location-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
    <strong>Motivation:</strong>
        Given our definition of clusters (above) identify them along with the
        "known" and "candidate, \lt 1MB" genes.  These gene sets will be
        combined with the observed ranking for more detailed analysis.
    </p>
    <p>
    <strong>Result:</strong>
        Altogether 60 "known" and 701 "candidate, \lt 1MB" genes
        before quality filtering.  These form 36 clusters, each with at least a few genes and a median of
        15 genes.  After filtering 36 "known" and 266 "candidate, \lt 1MB" genes.</p>
    <p>
    <strong>Interpretation:</strong>
        None.
    </p>
    <p>
    <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-08-08-imprinted-gene-clusters %}">
            2016-08-08-imprinted-gene-clusters
        </a>, <a href="{{ site.baseurl }}{% post_url 2016-07-14-imprinting-resources %}">
            2016-07-14-imprinting-resources 
        </a>
    </p>
</section>

<section>
    <h3>Top ranking genes</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-08-01-ifats-filters/top-ranking-genes-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-08-01-ifats-filters/top-ranking-genes-1.png">
        </a>
        <figcaption>
        top-genes
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-08-01-ifats-filters/top-ranking-genes-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
        <strong>Motivation:</strong>
        Based on gene score and clusters
        <ol>
            <li>
                detect novel imprinted genes
            </li>
            <li>define gene set for detailed subsequent analysis</li>
        </ol>
    </p>
    <p>
        <strong>Result:</strong>
        The figure presents the top ranking genes colored according to their
        imprinting category ("known",...).
    </p>
    <p>
        <strong>Interpretation:</strong>
        Evidence for some novel imprinted genes, e.g. TMEM261P1.  Several
        "known" imprinted genes are not among these top ranking genes
        suggesting these may not be imprinted in most human brains.
        Limitation: we did not estimate error rates.
    </p>
    <p>
        <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-08-01-ifats-filters %}">
            2016-08-01-ifats-filters
        </a>, <a href="https://docs.google.com/presentation/d/1YvpA1AJ-zzir1Iw0F25tO9x8gkSAzqaO4fjB7K3zBhE/edit#slide=id.p4">
            Ifat's version
        </a>
    </p>
</section>

<section>
    <h3>Variation of parental bias with age</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-06-26-trellis-display-of-data/S-age-smooth-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-06-26-trellis-display-of-data/S-age-smooth-1.png" width=200>
        </a>
        <figcaption>
        S-age-institution
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-06-26-trellis-display-of-data/S-age-smooth-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
    <strong>Motivation:</strong>
    The observed variability across individuals calls for explanation.
    Several variables were recorded including age, gender and disease status.
        Imprinting: clearly important for perinatal developement but how about
        later stages and aging?
    </p>
    <p>
    <strong>Result:</strong>
        Age seems to affect both the mean and variance of S for some genes
        either by decreasing (ZNF331) or increasing (DIRAS3), but not others
        (MEST). Smoothing with local regression (LOESS) supports this.
    </p>
    <p>
    <strong>Interpretation:</strong>
    The result appears to suggest dependence of parental bias on age.
    However, age seems to be associated with the brain-collecting(?) institution.  Possible association with other
        variables may further complicate interpretation.
    </p>
    <p>
    <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-06-26-trellis-display-of-data %}">
            2016-06-26-trellis-display-of-data
        </a>
    </p>
</section>

<section class=supplementary>
    <h3>Pairwise associations among predictors</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-06-26-trellis-display-of-data/evar-scatterplot-matrix-2.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-06-26-trellis-display-of-data/evar-scatterplot-matrix-2.png" width=200>
        </a>
        <figcaption>
        predictor-associations
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-06-26-trellis-display-of-data/evar-scatterplot-matrix-2.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
    <strong>Motivation:</strong>
        Uncover association among putative predictors of parental bias.  In theory higher order
        associations are of interest but here only pairwise ones are studied,
        in particular those involving age of death.
    </p>
    <p>
    <strong>Result:</strong>
        Association of age is strong to institution, disease, RNA quality (RIN) and
        ancestry and is also substantial to gender.  Institution is also
        strongly associated
        to most variables. </p>
    <p>
    <strong>Interpretation:</strong>
        The latter finding suggests institution-specific technical procedures
        and sampling bias.
    </p>
    <p>
    <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-06-26-trellis-display-of-data %}">
            2016-06-26-trellis-display-of-data
        </a>
    </p>
</section>

<section class=supplementary>
    <h3>The amount of data</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-06-26-trellis-display-of-data/S-age-tot-read-count-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-06-26-trellis-display-of-data/S-age-tot-read-count-1.png" width=200>
        </a>
        <figcaption>
        amount-of-data
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-06-26-trellis-display-of-data/S-age-tot-read-count-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
    <strong>Motivation:</strong>
        TODO
    </p>
    <p>
    <strong>Result:</strong>
        TODO
    </p>
    <p>
    <strong>Interpretation:</strong>
        TODO
    </p>
    <p>
    <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-06-26-trellis-display-of-data %}">
            2016-06-26-trellis-display-of-data
        </a>
    </p>
</section>

<section class=supplementary>
    <h3>Regression models: predicted curves</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-04-22-glm-for-s-statistic/s-stat-cmp-simple-multiple-regr-peg3-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-04-22-glm-for-s-statistic/s-stat-cmp-simple-multiple-regr-peg3-1.png">
        </a>
        <figcaption>
        predicted-curves
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-04-22-glm-for-s-statistic/s-stat-cmp-simple-multiple-regr-peg3-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
    <strong>Motivation:</strong>
        Blah-blah-blah
    </p>
    <p>
    <strong>Result:</strong>
        Blah-blah-blah
    </p>
    <p>
    <strong>Interpretation:</strong>
        Blah-blah-blah
    </p>
    <p>
    <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-04-22-glm-for-s-statistic %}">
            2016-04-22-glm-for-s-statistic
        </a>
    </p>
</section>

<section class=supplementary>
    <h3>Normal linear model fits well only to transformed data</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-05-07-comparing-regression-models/s-r-stat-nlm-check-peg3-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-05-07-comparing-regression-models/s-r-stat-nlm-check-peg3-1.png">
        </a>
        <figcaption>
        non-constant-variance
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-05-07-comparing-regression-models/s-r-stat-nlm-check-peg3-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
    <strong>Motivation:</strong>
        Blah-blah-blah
    </p>
    <p>
    <strong>Result:</strong>
        Blah-blah-blah
    </p>
    <p>
    <strong>Interpretation:</strong>
        Blah-blah-blah
    </p>
    <p>
    <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-05-07-comparing-regression-models %}">
            2016-05-07-comparing-regression-models
        </a>
    </p>
</section>

<section>
    <h3>Age effect and gene clusters</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/segplot-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/segplot-1.png">
        </a>
        <figcaption>
        age-effect
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/segplot-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
    <strong>Motivation:</strong>
        Blah-blah-blah
    </p>
    <p>
    <strong>Result:</strong>
        Blah-blah-blah
    </p>
    <p>
    <strong>Interpretation:</strong>
        Age operates on genes within the same cluster independently rather
        than in concert.  The wide confidence intervals and the relatively
        poor model fit weaken this conclusion.
    </p>
    <p>
    <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-08-08-imprinted-gene-clusters %}">
            2016-08-08-imprinted-gene-clusters
        </a>
    </p>
</section>

<section class=supplementary>
    <h3>All regression coefficients</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-06-22-extending-anova/reg-coef-logi.S-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-06-22-extending-anova/reg-coef-logi.S-1.png" width=200>
        </a>
        <figcaption>
        all-effects
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-06-22-extending-anova/reg-coef-logi.S-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
    <strong>Motivation:</strong>
        TODO
    </p>
    <p>
    <strong>Result:</strong>
        TODO
    </p>
    <p>
    <strong>Interpretation:</strong>
        TODO
    </p>
    <p>
    <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-06-22-extending-anova %}">
            2016-06-22-extending-anova
        </a>
    </p>
</section>

<section class=supplementary>
    <h3>All regression coefficients (wnlm.R model)</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-06-22-extending-anova/reg-coef-logi.S-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-06-22-extending-anova/reg-coef-logi.S-1.png" width=200>
        </a>
        <figcaption>
        all-effects-wnlm.R
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-06-22-extending-anova/reg-coef-logi.S-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
    <strong>Motivation:</strong>
        TODO
    </p>
    <p>
    <strong>Result:</strong>
        TODO
    </p>
    <p>
    <strong>Interpretation:</strong>
        TODO
    </p>
    <p>
    <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-06-22-extending-anova %}">
            2016-06-22-extending-anova
        </a>
    </p>
</section>

<section class=supplementary>
    <h3>Comparison of age effect under different regression models</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-06-17-extending-regression-analysis/beta-age-wnlm.R-vs-logi.S-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-06-17-extending-regression-analysis/beta-age-wnlm.R-vs-logi.S-1.png" width=200>
        </a>
        <figcaption>
        age-effect-two-models
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-06-17-extending-regression-analysis/beta-age-wnlm.R-vs-logi.S-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
    <strong>Motivation:</strong>
        TODO
    </p>
    <p>
    <strong>Result:</strong>
        TODO
    </p>
    <p>
    <strong>Interpretation:</strong>
        TODO
    </p>
    <p>
    <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-06-17-extending-regression-analysis %}">
            2016-06-17-extending-regression-analysis
        </a>
    </p>
</section>

<section>
    <h3>Non-orthogonality obscures relative impact of predictors</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-06-22-extending-anova/anova-effects-fw-rv-logi.S-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-06-22-extending-anova/anova-effects-fw-rv-logi.S-1.png" width=200>
        </a>
        <figcaption>
        anova
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-06-22-extending-anova/anova-effects-fw-rv-logi.S-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
    <strong>Motivation:</strong>
        TODO
    </p>
    <p>
    <strong>Result:</strong>
        TODO
    </p>
    <p>
    <strong>Interpretation:</strong>
        TODO
    </p>
    <p>
    <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-06-22-extending-anova %}">
            2016-06-22-extending-anova
        </a>
    </p>
</section>

<section class=supplementary>
    <h3>Interaction of age with other predictors</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-07-08-conditional-inference/beta-age-cond-logi.S-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-07-08-conditional-inference/beta-age-cond-logi.S-1.png" width=200>
        </a>
        <figcaption>
        interaction
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-07-08-conditional-inference/beta-age-cond-logi.S-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
    <strong>Motivation:</strong>
        TODO
    </p>
    <p>
    <strong>Result:</strong>
        TODO
    </p>
    <p>
    <strong>Interpretation:</strong>
        TODO
    </p>
    <p>
    <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-07-08-conditional-inference %}">
            2016-07-08-conditional-inference
        </a>
    </p>
</section>

<section class=missing>
    <h3>Comparison to previous mouse study: genome-wide parental bias</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/beta-age-logi.S-braim.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/beta-age-logi.S-braim.png" width=200>
        </a>
        <figcaption>
        TODO
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/beta-age-logi.S-braim.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
    <strong>Motivation:</strong>
        TODO
    </p>
    <p>
    <strong>Result:</strong>
        TODO
    </p>
    <p>
    <strong>Interpretation:</strong>
        TODO
    </p>
    <p>
    <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-08-08-imprinted-gene-clusters %}">
            2016-08-08-imprinted-gene-clusters
        </a>
    </p>
</section>

<section class=missing>
    <h3>Comparison to previous mouse study: age effect</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/beta-age-logi.S-braim.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/beta-age-logi.S-braim.png" width=200>
        </a>
        <figcaption>
        TODO
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/beta-age-logi.S-braim.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
    <strong>Motivation:</strong>
        TODO
    </p>
    <p>
    <strong>Result:</strong>
        TODO
    </p>
    <p>
    <strong>Interpretation:</strong>
        TODO
    </p>
    <p>
    <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-08-08-imprinted-gene-clusters %}">
            2016-08-08-imprinted-gene-clusters
        </a>
    </p>
</section>

