---
layout: page
title: Monoall-ms
---

<link rel="stylesheet" href="{{ "/css/manuscript-summary.css" | prepend: site.baseurl }}">

<h2>Downloads</h2>

Get the manuscript in
<a href="{{ site.baseurl }}/monoall-ms/monoall-ms.rtf">rtf</a>
or
<a href="{{ site.baseurl }}/monoall-ms/monoall-ms.pdf">pdf</a>
format, or download the
<a href="{{ site.baseurl }}/monoall-ms/monoall-ms.tex">latex</a>
and
<a href="{{ site.baseurl }}/monoall-ms/monoall-ms.bib">bibtex</a>
source files.

<h2>Variation of Genomic Imprinting in the Human Population,
and Analysis of its Sources and Psychiatric Consequences</h2>

<section>
    <h3>Preliminaries</h3>
    <h4>Notation</h4>
    <p><strong>Color key</strong>
        <ul>
            <li> <dfn>normal color:</dfn> text related to main figure</li>
            <li> <dfn class=supplementary>first color:</dfn> ...supplementary figure</li>
            <li> <dfn class=missing>second color:</dfn> ...figure still missing</li>
        </ul>
    </p>
    <p> <strong>refname:</strong> this should be used as unique identifier for
        a figure until the final figure numbers may be established</p>
    <h4>Definitions</h4>
    <ul>
        <li>
            <dfn>monoallelic expression:</dfn> only one allele is expressed to a
            certain degree
        </li><li>
            <dfn>(genomic) imprinting:</dfn> a subtype of monoallelic expression
            where the same allele is expressed in some tissue across all
            individuals
        </li><li>
            <dfn>known imprinted</dfn> and <dfn>candidate</dfn> genes: with
            and without prior evidence for imprinting, respectively
        </li><li>
            <dfn>nearby candidate</dfn> a candidate gene in the 1 MB
            neighborhood of some known imprinted gene
        </li><li>
            <dfn>imprinted gene cluster:</dfn> a sequence of neighboring genes
            that fall in either the "known" or the "candidate, \lt 1 MB gene"
            category
        </li><li>
            <dfn>biallelic expression:</dfn> both alleles are expressed to a
            certain degree
        </li><li>
            <dfn>unbiased, or balanced, expression:</dfn> the two alleles are
            expressed at equally high level
        </li><li>
            <dfn>parental expression bias</dfn> or simply <dfn>parental bias:</dfn> one allele is expressed at a
            higher level
        </li><li>
            <dfn>mean transcript ratio, \mu_{ig}</dfn> of a given individual i
            and gene g:
            <ol>
                <li>the mean ratio of the number of maternal transcripts in
                    all transcripts over a suitable period of time and set of
                    cells
                </li><li>hence a natural but not directly unobservable measure of expression bias</li>
            </ol>
        </li>
    </ul>
</section>

<section>
    <h3>Parental expression bias: variation across genes and individuals</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-07-19-genome-wide-S/complex-plot-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-07-19-genome-wide-S/complex-plot-1.png">
        </a>
        <figcaption>
        ranking-genes
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-07-19-genome-wide-S/complex-plot-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
        <strong>Motivation:</strong>
        Genome-wide and population-wide quantification parental bias
        <ol>
            <li>variability across genes and the fraction of monoallelically
                expressed genes</li>
            <li>variability across individuals</li>
        </ol>
        We define<dfn>read count ratio statistic, S_{ig}</dfn> of a given
        individual i and gene g as
        the greater of the observed read count of the two alleles at each
        heterozygous SNP summed over all such SNPs and normalized to the
        corresponding total read count.  We use S_{ig} to quantify parental
        bias.  This is justified by the reasonable assumptions that
        <ol>
            <li>the number of paternal and maternal transcripts are Poisson
                distributed with rate \lambda and \lambda \mu, respectively
            </li>
            <li>the observed paternal and maternal read count are both
                Poisson distributed with rate proportional to the number of
                paternal and maternal transcripts, respectively.
            </li>
        </ol>
        We also define the <dfn>gene g's score</dfn> as the fraction of
        individuals i for which S_{ig}\gt 0.9.  This is the same as 1 -
        ECDF(0.9), where ECDF is the empirical cumulative distribution
        function based on observations S_{1g},...,S_{Ig} on I individuals or,
        equivalently, the same as the corresponding survival function.
    </p>
    <p>
        <strong>Result:</strong>
        Three typical genes are shown in the <strong>upper 3 panels</strong>:
        PEG10 and ZNF331 are known imprinted genes, whereas AFAP1 is a
        candidate gene.  The each gene has a characteristic survival function
        1 - ECDF(s) based on which receive different scores <strong>green
            circles a, b, c</strong>.  The survival functions are color-coded
        for terse representation.  All ca. 5K genes that passed our quality
        filters are shown in this representation<strong>2 main lower
            panels</strong>; they are ranked according to their score, shown
        in the auxiliary <strong>right lower panels</strong>.
    </p>
    <p>
        <strong>Interpretation:</strong>
        Given some reasonable assumptions (see definition of read count ratio
        statistic S) only about ca. 1 % of the genes are expressed
        monoallelically
    </p>
    <p>
        <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-07-19-genome-wide-S %}">
            2016-07-19-genome-wide-S
        </a>
    </p>
</section>

<section class=supplementary>
    <h3>Parental bias and imprinted gene clusters</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/score-genomic-location-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/score-genomic-location-1.png">
        </a>
        <figcaption>
        clusters
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/score-genomic-location-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
        <strong>Motivation:</strong>
        Given our definition of clusters (above) identify them along with the
        "known" and "candidate, \lt 1MB" genes.  These gene sets will be
        combined with the observed ranking for more detailed analysis.
    </p>
    <p>
        <strong>Result:</strong>
        Altogether 60 "known" and 701 "candidate, \lt 1MB" genes
        before quality filtering.  These form 36 clusters, each with at least a few genes and a median of
        15 genes.  After filtering 36 "known" and 266 "candidate, \lt 1MB" genes.</p>
    <p>
        <strong>Interpretation:</strong>
        None.
    </p>
    <p>
        <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-08-08-imprinted-gene-clusters %}">
            2016-08-08-imprinted-gene-clusters
        </a>, <a href="{{ site.baseurl }}{% post_url 2016-07-14-imprinting-resources %}">
            2016-07-14-imprinting-resources 
        </a>
    </p>
</section>

<section>
    <h3>Top ranking genes</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-08-01-ifats-filters/top-ranking-genes-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-08-01-ifats-filters/top-ranking-genes-1.png">
        </a>
        <figcaption>
        top-genes
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-08-01-ifats-filters/top-ranking-genes-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
        <strong>Motivation:</strong>
        Based on gene score and clusters
        <ol>
            <li>
                detect novel imprinted genes
            </li>
            <li>define gene set for detailed subsequent analysis</li>
        </ol>
    </p>
    <p>
        <strong>Result:</strong>
        The figure presents the top ranking genes colored according to their
        imprinting category ("known",...).
    </p>
    <p>
        <strong>Interpretation:</strong>
        Evidence for some novel imprinted genes, e.g. TMEM261P1.  Several
        "known" imprinted genes are not among these top ranking genes
        suggesting these may not be imprinted in most human brains.
        Limitation: we did not estimate error rates.
    </p>
    <p>
        <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-08-01-ifats-filters %}">
            2016-08-01-ifats-filters
        </a>, <a href="https://docs.google.com/presentation/d/1YvpA1AJ-zzir1Iw0F25tO9x8gkSAzqaO4fjB7K3zBhE/edit#slide=id.p4">
            Ifat's version
        </a>
    </p>
</section>

<section>
    <h3>Variation of parental bias with age</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-06-26-trellis-display-of-data/S-age-gender-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-06-26-trellis-display-of-data/S-age-gender-1.png" width=200>
        </a>
        <figcaption>
        S-age-gender
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-06-26-trellis-display-of-data/S-age-gender-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
        <strong>Motivation:</strong>
        The observed variability across individuals calls for explanation.
        Several variables were recorded including age, gender and disease status.
        Imprinting: clearly important for perinatal developement but how about
        later stages and aging?
    </p>
    <p>
        <strong>Result:</strong>
        Age seems to affect both the mean and variance of S for some genes
        either by decreasing (ZNF331) or increasing (DIRAS3), but not others
        (MEST).
    </p>
    <p>
        <strong>Interpretation:</strong>
        The result appears to suggest dependence of parental bias on age.
    </p>
    <p>
        <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-06-26-trellis-display-of-data %}">
            2016-06-26-trellis-display-of-data
        </a>
    </p>
</section>

<section class=supplementary>
    <h3>Pairwise associations among predictors</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-06-26-trellis-display-of-data/evar-scatterplot-matrix-2.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-06-26-trellis-display-of-data/evar-scatterplot-matrix-2.png" width=200>
        </a>
        <figcaption>
        predictor-associations
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-06-26-trellis-display-of-data/evar-scatterplot-matrix-2.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
        <strong>Motivation:</strong>
        Uncover association among putative predictors of parental bias.  In theory higher order
        associations are of interest but here only pairwise ones are studied,
        in particular those involving age of death.
    </p>
    <p>
        <strong>Result:</strong>
        Association of age is strong to institution, disease, RNA quality (RIN) and
        ancestry and is also substantial to gender.  Institution is also
        strongly associated
        to most variables. </p>
    <p>
        <strong>Interpretation:</strong>
        The latter finding suggests institution-specific technical procedures
        and sampling bias.
    </p>
    <p>
        <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-06-26-trellis-display-of-data %}">
            2016-06-26-trellis-display-of-data
        </a>
    </p>
</section>

<section class=supplementary>
    <h3>Weight of evidence</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-06-26-trellis-display-of-data/S-age-tot-read-count-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-06-26-trellis-display-of-data/S-age-tot-read-count-1.png" width=200>
        </a>
        <figcaption>
        weight-of-evidence
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-06-26-trellis-display-of-data/S-age-tot-read-count-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
        <strong>Motivation:</strong>
        The total read count is informative for statistical inference on the
        effect of age and other predictors since the latter may be considered as a
        weigh of evidence on the read count ratio for each data point that corresponds
        to some individual--gene pair.
    </p>
    <p>
        <strong>Result:</strong>
        The total read count varies both across genes and, for a given gene,
        across individuals.  Summing the higher and total read count over
        genes yields a weighted average (WA) of the read count ratio.
    </p>
    <p>
        <strong>Interpretation:</strong>
        The across-genes variation of read count may chiefly reflect across-genes
        variation in expression, whereas the across-individuals variation of
        read count likely arises from variation both in expression and in
        technological effects.  Modeling these levels of
        variation is challenging but desirable since that affords higher
        sensitivity and specificity.
    </p>
    <p>
        <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-06-26-trellis-display-of-data %}">
            2016-06-26-trellis-display-of-data
        </a>
    </p>
</section>

<section>
    <h3>Regression models</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-08-23-glm-sampling-distributions/ZDBF2-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-08-23-glm-sampling-distributions/ZDBF2-1.png">
        </a>
        <figcaption>
        predicted-curves
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-08-23-glm-sampling-distributions/ZDBF2-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
        <strong>Motivation:</strong>
        Fitting a suitable regression model is the standard technique to
        quantitatively infer putative predictors' effect on a response
        variable, in this case the read count ratio.  We need a regression
        model that best accounts for our observation of associations among
        predictors and of varying weight of evidence.  We tested several
        models that differ in the following characteristics:
        <ol>
            <li>data transformation (none or rank tr.)</li>
            <li>weighting according to total read count (weighted or
                unweighted)</li>
            <li>link function and error distribution (logistic and normal
                linear)</li>
            <li>relationship among predictors (simple or multiple regression)</li>
            <li>relationship among genes (independent or identically
                distributed).</li>
        </ol>
        The logistic models have several theoretical advantages
        over the normal linear models besides keeping predicted ratios below 1
        <ul>
            <li>
                they model the higher read count as binomial variables, which is not
                only more reasonable than normality but provides a natural way of
                using total read counts as weights of evidence </li>
            <li>they model the strongly unequal error variance observed in the
            data.</li>
        </ul>
    </p>
    <p>
        <strong>Result:</strong>
        In summary fitting various models revealed the similar qualitative
        trends.  The figure shows that normal linear models predict, with
        positive probability, read count ratios greater than 1 and fail to
        account for the systematic increase of variance with decreasing
        average read count ratio.  The rank transformation achieves nearly
        constant variance but looses information and thereby substantially
        worsens signal-to-noise (explained vs unexplained variance).
        Logistic models capture patterns of data much better, although they
        exlain a more moderate change in variance than what is observed
        (overdispersion).  However, the steepest,
        most informative, part of their sigmoidal predicted curve lies out of
        the observed range of data (massive extrapolation).  The predicions
        under the unweighted norlal linear models (unlm.S and unlm.R) are very
        similar to their weighted counterparts (wnlm.S and wnlm.R) and are not
        shown.
    </p>
    <p>
        <strong>Interpretation:</strong>
        As expected based on theoretical grounds logistic models indeed seem
        to express several properties of the data a lot better than normal linear
        models.  The data seems overdispersed indicating that it would be
        beneficial to use hierachical models whose layers correspond to
        various sources of biological and technical variation.  The former
        includes fluctuation in gene expression (across time and cells) as
        well as variability and shared information across genes.
    </p>
    <p>
        <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-08-23-glm-sampling-distributions %}">
            2016-08-23-glm-sampling-distributions
        </a>
    </p>
        <a href="{{ site.baseurl }}{% post_url 2016-04-22-glm-for-s-statistic %}">
            2016-04-22-glm-for-s-statistic
        </a>
    </p>
</section>

<section class=supplementary>
    <h3>Normal linear model fits well only to transformed data</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-05-07-comparing-regression-models/s-r-stat-nlm-check-peg3-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-05-07-comparing-regression-models/s-r-stat-nlm-check-peg3-1.png">
        </a>
        <figcaption>
        non-constant-variance
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-05-07-comparing-regression-models/s-r-stat-nlm-check-peg3-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
        <strong>Motivation:</strong>
        Normal linear models become strongly biased when fitted to data with
        non-constant error variance.
    </p>
    <p>
        <strong>Result:</strong>
        The error variance is strongly varying: both systematically (with age,
        for instance) as seen in the <strong>top left</strong> panel and
        non-systematically (several outlier data points) as shown by the
        <strong>top right</strong> panel.  The <strong>bottom</strong> panels
        were obtained with rank transformation and show much improvement due
        to reduced systematic and non-systematic variation.
    </p>
    <p>
        <strong>Interpretation:</strong>
        The result is in line with the theoretical disadvantages of the normal
        linear model and suggests that inference from this model should not be
        taken into account because the inferred parameters are strongly
        biased.  Rank transformation is likely to remove most of the bias at
        the expense of reduced sensitivity and interpretability since it
        places read count ratio on relative scale.
    </p>
    <p>
        <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-05-07-comparing-regression-models %}">
            2016-05-07-comparing-regression-models
        </a>
    </p>
</section>

<section>
    <h3>Biological predictors and gene clusters</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/segplot-wnlm-Q-99conf-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/segplot-wnlm-Q-99conf-1.png">
        </a>
        <figcaption>
        biol-effects
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/segplot-wnlm-Q-99conf-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
        <strong>Motivation:</strong>
        Given the results and theoretical considerations the logistic multiple
        regression model is used to infer \beta_\mathrm{age}, the regression
        coefficient that mediates the effect of age on read count ratio.
    </p>
    <p>
        <strong>Result:</strong>
        When genes are taken independent, a separate \beta_\mathrm{age} is
        inferred for each of them.  This shows great variability across genes:
        at 99 % confidence
        some have significantly negative, others significantly positive
        coefficient and yet others a coefficient not significantly different
        from zero.  The variability is also seen within imprinted gene
        clusters.  Compared to the logistic model (shown) the normal linear
        model with rank transformation appears much less powerful
        (significance only for 1 gene: ZDBF2).  Decreasing confidence to 95 % improves
        power (significance for 5 genes).
    </p>
    <p>
        <strong>Interpretation:</strong>
        Some genes loose and others gain parental expression bias with age,
        while a third set of genes are not affected by age.  An alternative,
        not necessary exclusive, interpretation is that the extrapolation
        under the logistic model leads to biased coefficients.  The
        qualitative agreement between the two models supports the biological
        interpretation.  If the age effect is indeed true then age operates on
        genes within the same cluster independently rather than in concert.
    </p>
    <p>
        <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-08-08-imprinted-gene-clusters %}">
            2016-08-08-imprinted-gene-clusters
        </a>
    </p>
</section>

<section class=supplementary>
    <h3>All regression coefficients</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-06-22-extending-anova/reg-coef-logi-S-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-06-22-extending-anova/reg-coef-logi-S-1.png" width=200>
        </a>
        <figcaption>
        all-effects-logi.S
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-06-22-extending-anova/reg-coef-logi-S-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
        <strong>Motivation:</strong>
        Apart from age, we have data on other biological explanatory
        variables, specifically disease status, gender and ancestry.  Do these also modulate
        parental bias like age seems to do?
    </p>
    <p>
        <strong>Result:</strong>
        Gender and some principal components of the ancestry have significant
        effects for several genes.  In contrast, disease status (Dx) does not,
        except for a single gene, NDN.
    </p>
    <p>
        <strong>Interpretation:</strong>
        The effect of age, at least on some genes, seems clear despite the
        limitations of the logistic model.  The significance of other
        biological explanatory variables is unclear because of the observed sources of
        bias and high noise associated with technical variables.
    </p>
    <p>
        <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-06-22-extending-anova %}">
            2016-06-22-extending-anova
        </a>
    </p>
</section>

<section class=supplementary>
    <h3>All regression coefficients (wnlm.R model)</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-06-22-extending-anova/reg-coef-wnlm-R-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-06-22-extending-anova/reg-coef-wnlm-R-1.png" width=200>
        </a>
        <figcaption>
        all-effects-wnlm.R
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-06-22-extending-anova/reg-coef-wnlm-R-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
        <strong>Motivation:</strong>
        Does the qualitative agreement between the logistic model logi.S (fitted on read
        count ratios without transformation) and normal linear model wnlm.R (fitted
        after rank-transformation) observed for age's effect also hold for
        other predictors.
    </p>
    <p>
        <strong>Result:</strong>
        Similar qualitative patterns are found under wnlm.R as under logi.S.
        In general, the confidence intervals are relatively broader.
    </p>
    <p>
        <strong>Interpretation:</strong>
        These results provide further support for the significant effects of
        some of the biological variables explaining parental bias.  wnlm.R is
        less powerful (sensitive) than logi.S.
    </p>
    <p>
        <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-06-22-extending-anova %}">
            2016-06-22-extending-anova
        </a>
    </p>
</section>

<section class=supplementary>
    <h3>Comparison of age effect under different regression models</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-06-17-extending-regression-analysis/beta-age-wnlm-R-vs-logi-S-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-06-17-extending-regression-analysis/beta-age-wnlm-R-vs-logi-S-1.png" width=200>
        </a>
        <figcaption>
        age-effect-two-models
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-06-17-extending-regression-analysis/beta-age-wnlm-R-vs-logi-S-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
        <strong>Motivation:</strong>
        A more direct comparison of the logistic (logi.S) and normal linear
        models (wnlm.R) may help more firmly establish the preceding results.
    </p>
    <p>
        <strong>Result:</strong>
        Clearly, the two models agree qualitatively on age's effect.
    </p>
    <p>
        <strong>Interpretation:</strong>
        The two models quite different in several statistical properties.
        Thus, relatively few shared sources of bias are possible making the
        biological interpretation of the significant effects more plausible.
        In what follows we turn to an obvious shared property of the two
        models: (generalized) linearity.
    </p>
    <p>
        <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-06-17-extending-regression-analysis %}">
            2016-06-17-extending-regression-analysis
        </a>
    </p>
</section>

<section class=supplementary>
    <h3>Likelihood: association of regression coefficients</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-08-21-likelihood-surface/ll-surf-coefs-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-08-21-likelihood-surface/ll-surf-coefs-1.png" width=200>
        </a>
        <figcaption>
        likelihood-surface
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-08-21-likelihood-surface/ll-surf-coefs-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
        <strong>Motivation:</strong>
        The observed associations (dependencies) among predictors are expected to induce associations
        among the coefficients $\{\beta_i\}$ themselves, which in turn makes MLE
        less reliable.  In the extreme case of collinearity coefficients become
        unidentifiable and MLE fails.  Pairwise associations involving
        \beta_\mathrm{Age} and other coefficients are investigated below.
    </p>
    <p>
        <strong>Result:</strong>
        \beta_\mathrm{Age} is associated with coefficients of both technical and biological predictors.  Association to the coefficient of RNA-quality RIN (RNA Integrity Number) is the strongest.
    </p>
    <p>
        <strong>Interpretation:</strong>
        These associations, especially the strongest ones, caution about the
        inferences made with maximum likelihood criterion.  From these pairwise
        associations the network of causal dependencies does not follow directly and
        neither does the violation of the assumption of linearity.
    </p>
    <p>
        <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-08-21-likelihood-surface %}">
            2016-08-21-likelihood-surface
        </a>
    </p>
</section>

<section>
    <h3>Associations obscure the size of the age effect</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-06-22-extending-anova/anova-effects-fw-rv-logi-S-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-06-22-extending-anova/anova-effects-fw-rv-logi-S-1.png" width=200>
        </a>
        <figcaption>
        anova
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-06-22-extending-anova/anova-effects-fw-rv-logi-S-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
        <strong>Motivation:</strong>
        Before more directly inspecting linearity we study the possibility of
        near collinearity of sets of predictors.  Orthogonality (lack of
        collinearity) is desirable to make statements on the relative impact
        of predictors and therefore the biological-"signal"-to-technical-"noise"
        ratio.  In particular we ask: how much change in parental bias can
        aging induce? This can be studied with ANOVA as well as with the
        decomposition of the response response vector along regression
        coefficients (observed mediators of effects) and unobserved mediators
        of effects.
    </p>
    <p>
        <strong>Result:</strong>
        When the age component is taken first into account the corresponding
        deviance and effect is roughly as high as the highest technical
        variables (Institution and RNA_batch).  But in the reverse direction
        the age component greatly shrinks.
    </p>
    <p>
        <strong>Interpretation:</strong>
        The shrinkage is due to age's association to institution and other
        predictors, as we found above.
        This does not weaken the previously observed significance of
        age's effect but it does preclude determination of age's effect size.
    </p>
    <p>
        <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-06-22-extending-anova %}">
            2016-06-22-extending-anova
        </a>
    </p>
</section>

<section class=supplementary>
    <h3>Interaction of age with other predictors</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-07-08-conditional-inference/beta-age-cond-logi-S-1.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-07-08-conditional-inference/beta-age-cond-logi-S-1.png" width=200>
        </a>
        <figcaption>
        interaction
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-07-08-conditional-inference/beta-age-cond-logi-S-1.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
        <strong>Motivation:</strong>
        Linearity is a shared assumption by all models used here, whose
        violation might bias inference on the effect of age and other
        predictors.  Linearity is equivalent to the conditional independence
        of predictors given the response (i.e. read count ratio).
    </p>
    <p>
        <strong>Result:</strong>
        Conditioning the inference of \beta_\mathrm{age} on institution and
        gender seems to more strongly affect inferences under the logistic model logi.S
        (shown) than those under the normal linear model wnlm.R.  The pattern
        of change in the inferred \beta_\mathrm{age} is more often similar
        than different.
    </p>
    <p>
        <strong>Interpretation:</strong>
        The result suggests substantial nonlinearity; in other words age
        interacts with institution and gender.  This interaction is not
        accounted for by our models, which may potentially lead to quit strong
        bias. The normal linear model (fitted to rank-transformed read counts)
        is more robust to departures from linearity.  This points to the
        importance of comparing our observational study with factorially
        designed experiments.
    </p>
    <p>
        <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-07-08-conditional-inference %}">
            2016-07-08-conditional-inference
        </a>
    </p>
</section>

<section class=missing>
    <h3>Comparison to previous mouse study: genome-wide parental bias</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/beta-age-logi-S-braim.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/beta-age-logi-S-braim.png" width=200>
        </a>
        <figcaption>
        TODO
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/beta-age-logi-S-braim.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
        <strong>Motivation:</strong>
        TODO
    </p>
    <p>
        <strong>Result:</strong>
        TODO
    </p>
    <p>
        <strong>Interpretation:</strong>
        TODO
    </p>
    <p>
        <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-08-08-imprinted-gene-clusters %}">
            2016-08-08-imprinted-gene-clusters
        </a>
    </p>
</section>

<section class=missing>
    <h3>Comparison to previous mouse study: age effect</h3>
    <figure>
        <a href="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/beta-age-logi-S-braim.png">
            <img src="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/beta-age-logi-S-braim.png" width=200>
        </a>
        <figcaption>
        TODO
        <p>
            <a href="{{ site.baseurl }}/monoall-ms/2016-08-08-imprinted-gene-clusters/beta-age-logi-S-braim.pdf">pdf</a> 
        </p>
        </figcaption>
    </figure>
    <p>
        <strong>Motivation:</strong>
        TODO
    </p>
    <p>
        <strong>Result:</strong>
        TODO
    </p>
    <p>
        <strong>Interpretation:</strong>
        TODO
    </p>
    <p>
        <strong>See also:</strong>
        <a href="{{ site.baseurl }}{% post_url 2016-08-08-imprinted-gene-clusters %}">
            2016-08-08-imprinted-gene-clusters
        </a>
    </p>
</section>

